File: twosum.ts
const nums = [2, 7, 11, 15];

const target = 9;
function twoSum(nums: number[], target: number): number[] {
  const seen = new Map<number, number>();

  for (let i = 0; i < nums.length; i++) {
    const x = nums[i];
    const comp = target - x;

    if (seen.has(comp)) {
      return [seen.get(comp)!, i];
    }

    seen.set(x, i);
  }

  return [-1, -1];
}

console.log(twoSum(nums, target));

File: twosum.go
package main

import "fmt"




func twoSum(nums []int, target int) (int, int) {
    seen := make(map[int]int) 

    for i, x := range nums {
        comp := target - x

        if j, ok := seen[comp]; ok {
            return j, i 
        }

        seen[x] = i
    }
    return -1, -1 
}

	

func main() {
    nums := []int{2, 7, 11, 15}
    target := 9

    i, j := twoSum(nums, target)
    fmt.Println("Indices:", i, j) 
}